docs/wip/exchange-onboarding.md

```diff
@@ -0,0 +1,381 @@
+private relays:
+
+    TODO: private relays should be inserted here in document given to exchanges
+
+## Building and running
+
+We support two different ways to build and run the wallet. You can run
+it standalone on any linux system built in the nix store or you can generate
+a docker container that can be ran on any docker container orchestrator like
+docker swarm or kubernetes.
+
+### Build and run in the nix store
+
+By default the wallet's local state goes in
+`./state-wallet-mainnet`.
+
+Build the wallet and generate the shell script to connect to
+mainnet (use `connectScripts.stagingWallet` for testnet)
+
+    nix-build -A connectScripts.mainnetWallet -o "./launch_$(date -I)_$(git rev-parse --short HEAD)"
+
+After the build finishes the generated connection script is
+available as a symlink called `./launch_2018-01-30_0d4f79eea`, or
+similar. Run that symlink as a script to start the wallet.
+
+### Build and run a docker image
+
+Follow the above instructions for customization and dependencies. To build a docker
+container and import the image run
+(use `connectScripts.stagingWallet` for testnet):
+
+    docker load < $(nix-build -A dockerImages.mainnetWallet)
+
+This will create an image `cardano-container-mainnet:latest`
+(or `cardano-container-staging:latest` for testnet)
+
+After this image is built, it can be used like any other docker image being pushed
+into a registry and pulled down using your preferred docker orchestration tool.
+
+The image can be ran using the following:
+
+    docker run --name cardano-mainnet-wallet --rm -it -p 127.0.0.1:8090:8090 -p 127.0.0.1:8000:8000 -v state-wallet-mainnet:/wallet cardano-container-mainnet:latest
+
+The above command will create a docker volume named `state-wallet-mainnet` and will mount
+that to /wallet. Note: if no volume is mounted to `/wallet` the container startup
+script will refuse to execute `cardano-node` and the container will exit.
+
+The location of `/wallet` cannot be changed, but you can mount any kind of volume
+you want in that directory that docker supports.
+
+Note that if you give this a different name than is specified above, use the
+name you used for any future docker commands in examples in the document.
+
+# Migrating from V0 to V1 API
+
+TODO: add instructions for migrating to new API
+
+# Usage FAQs
+
+## What are recommended hardware/software requirements for exchange wallets?
+
+RAM: 8 GB for building, 4 GB for running
+
+CPU: Modern x86_64 processor
+
+Disk: SSD recommended of 100 GB size with option to allocate more in future
+years.
+
+Operating System: NixOS or CentOS 7 recommended, although any linux
+distribution should work with nix package manager used for building
+the binaries
+
+Software Requirements: Nix package manager for building standalone
+and docker containers. On systems using docker, docker > 17.12 required.
+
+## How do I export the CA certificate for the API?
+
+The certificate is generated inside the wallet in the file `tls/server.cert`
```

Comment:
Hey Alfredo, it shouldn't I believe since the client's identity is embedded in the certificate. But please, see my comment on CBR-82 on that matter, I'd appreciate some inputs here :)

Reply:
@adinapoli-iohk @KtorZ 

I can definitely weigh in here coming from a security background in my last job. A client certificate needs to be signed by the CA. What I would recommend is having two separate workflows:

1) Daedalus: Generate an SSL CA on first run. Generate cert/key for server (cardano-sl) and cert/key for daedalus. Daedalus should use the client certificate to authenticate to cardano-sl. Optionally, include in daedalus UI a way to export the client cert/key for expert users that want to hit the API directly.

2) Exchanges: Add option for specifying a CA certificate in custom-wallet-config.nix. On first run, cardano-sl should output a CSR and quit. The exchange should then generate a certificate signed by the CA specified in custom-wallet-config.nix and put that certificate in the wallet directory. Then on subsequent runs, cardano-sl would use that server certificate. The client can then generate as many client certificates as they need to talk to the cardano-sl API.

If you want more details on CA setup and server/client certificate generation using a CA, I highly recommend https://jamielinux.com/docs/openssl-certificate-authority/index-full.html

Reply:
Hey @disassembler, perhaps you can have a look at #2683 & #2684. You inputs would be appreciated on the matter.

Reply:
@disassembler, here's some context on how it was implemented in Daedalus: https://iohk.myjetbrains.com/youtrack/issue/DEVOPS-33

In short, the CA is indeed generated at install-time.